### DISCUSSION TITLE
Wrong handling of % symbol

### DISCUSSION THREAD

**Discussion by sleivysarnas** [SwiftGen/SwiftGen#1016] [Q&A] [answered]
Hello,

I have two strings:
`"Please select a 0% selection:"` and `"This gives 50% off in discount"`

The first one is generated into `static let` variable, which is correct, but the second one is generated as a `func` accepting a parameter: 

`public static func percentageDiscount(_ p1: Int) -> String {}`

Both of them should be generated as variables. 

Any ideas on how to fix that? I have read about double percentage symbols `%%`, but why is it working correctly on the first string, it is with single percentage character too?

**Comment by djbe** [comment, marked as answer]
Because the first `% s` is not a format specifier, whereas `% o` is a valid format specifier. See [Format Specifiers](https://developer.apple.com/library/archive/documentation/CoreFoundation/Conceptual/CFStrings/formatSpecifiers.html#//apple_ref/doc/uid/TP40004265) for a list of format specifiers.

This question is a duplicate of quite a few existing questions, you can easily find them with GitHub search function.

**Comment by renmoqiqi** [comment]
1236

**Reply by renmoqiqi** [reply]
1236
